### PR TITLE
Syaam/20 medication schema

### PR DESCRIPTION
**Pull request by syaamkhandaker**
## PR Title

Issue Number(s): #20 .

What does this PR change and why?
Medication code

### Checklist

- [ ] Requirements have been implemented
- [ ] Acceptance criteria is met
- [ ] Database schema docs have been updated or are not necessary
- [ ] Code follows design and style guidelines
- [ ] Commits follow guidelines (concise, squashed, etc)
- [ ] Relevant reviewers (Senior Dev/EM/Designers) have been assigned to this PR

### Critical Changes

- Medication Schema
- Medication APi 
- Medication service layer

### Related PRs

- #

### Testing

Enumerate steps to test the functionality of your ticket. This should include edge cases and testing of error handling, if applicable.

Postman

### CODE REVIEW

File: src/pages/api/v1/medication/[id].ts
Comment (michaelhyi) [inline src/pages/api/v1/medication/[id].ts:1]: `GET` is returning an empty array while `PATCH` and `DELETE` are all throwing me errors on Postman. Please make sure to test all your endpoints thoroughly!

**Review by michaelhyi — CHANGES_REQUESTED** [review]
Great work Syaam! Please be sure to test all your endpoints. There were some errors from the deploy preview. Be sure to address all the comments I left! Great work!

File: src/pages/api/v1/medications/[userId].ts [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+import type { NextApiRequest, NextApiResponse } from "next";
+
+export default async function handler(
+  req: NextApiRequest,
+  res: NextApiResponse,
+) {
+  const { userId } = req.query;
+  const { method } = req;
+
+  if (typeof userId !== "string") {
```
[the remark this answers is not in the document]
Comment (syaamkhandaker) [reply src/pages/api/v1/medications/[userId].ts:13]: Yeah, I had written my code very similar to the other apis, so I figured I should include it too

[the remark this answers is not in the document]
Comment (syaamkhandaker) [reply src/pages/api/v1/medications/[userId].ts:13]: it can be another type though. the type is string | string[] | undefined

File: src/db/actions/medication.ts [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+  if (result.deletedCount == 0) {
+    throw new InternalServerError("Failed to delete medication.");
+  }
+}
+
+// this function retrieves list of medications
+export async function getMedicationsByUserId(userId: ObjectId) {
+  const db = client.db();
+
+  const medication = db.collection("medication").find({ userId });
```
Comment (michaelhyi) [inline src/db/actions/medication.ts:76]:
```suggestion
  const medication = await db.collection("medication").find({ userId });
```

**Review by michaelhyi — APPROVED** [review]
Yay! Works well
